### PR TITLE
API-1310: Create & dispatch ProductModelRemoved business event

### PR DESCRIPTION
**Pull request by pierrejolly**
**Description (for Contributor and Core Developer)**

Notify webhook when a product model has been deleted

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Component/Product/Webhook/ProductModelRemovedEventDataBuilder.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    public function build(BusinessEventInterface $businessEvent): array
+    {
+        if (false === $this->supports($businessEvent)) {
+            throw new \InvalidArgumentException();
+        }
+
+        $data = $businessEvent->data();
+
+        return [
+            'resource' => ['identifier' => $data['code']]
```
Comment (pchasle) [inline src/Akeneo/Pim/Enrichment/Component/Product/Webhook/ProductModelRemovedEventDataBuilder.php:34]:
```suggestion
            'resource' => ['code' => $data['code']]
```
To be compliant with the API format.

File: tests/back/Pim/Enrichment/Specification/Bundle/EventSubscriber/ProductModelEventSubscriberSpec.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    function it_does_not_produce_business_remove_event_because_event_subject_is_not_a_product(
+        $security,
+        $normalizer
+    ) {
+        $messageBus = $this->getMessageBus();
+        $this->beConstructedWith($security, $normalizer, $messageBus);
+
+        $result = $this->produceBusinessRemoveEvent(new GenericEvent('NOT_A_PRODUCT_MODEL'));
+
+        Assert::assertEquals(null, $result->getWrappedObject());
```
Comment (pchasle) [inline tests/back/Pim/Enrichment/Specification/Bundle/EventSubscriber/ProductModelEventSubscriberSpec.php:74]: I don't understand what you are testing here?
Should probably be `Assert::assertCount(0, $messageBus->messages);`

**Review by pchasle — APPROVED** [review]
Good; GTM after fixes

File: tests/back/Pim/Enrichment/Specification/Bundle/EventSubscriber/ProductModelEventSubscriberSpec.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+        $messageBus = $this->getMessageBus();
+        $this->beConstructedWith($security, $normalizer, $messageBus);
+
+        $user->getUsername()->willReturn('julia');
+        $security->getUser()->willReturn($user);
+
+        $normalizer->normalize($productModel, 'standard')->willReturn(
+            [
+                'code' => 12,
```
Comment (tomglvng) [inline tests/back/Pim/Enrichment/Specification/Bundle/EventSubscriber/ProductModelEventSubscriberSpec.php:56]:
```suggestion
                'code' => 'my_product_model',
```

File: tests/back/Pim/Enrichment/Specification/Bundle/EventSubscriber/ProductModelEventSubscriberSpec.php (hunk 2 above)
Comment (pierrejolly) [reply tests/back/Pim/Enrichment/Specification/Bundle/EventSubscriber/ProductModelEventSubscriberSpec.php:74]: Yes it's a copy, I will take a look on it

**Review by tomglvng — APPROVED** [review]

**Pull request merged**